packages/webcomponents/src/components/payment-details/test/payment-details.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import { newSpecPage } from "@stencil/core/testing";
+import { PaymentDetails } from "../payment-details";
+import { PaymentDetailsCore } from "../payment-details-core";
+import { Details } from "../../details/details";
+
+describe('payment-details', () => {
+  const components = [PaymentDetails, PaymentDetailsCore, Details];
+  it('renders an error message when paymentId and authToken are not provided', async () => {
+    const page = await newSpecPage({
+      components,
+      html: '<justifi-payment-details></justifi-payment-details>',
+    });
+    await page.waitForChanges();
+    expect(page.root).toMatchSnapshot();
```

Comment:
Very interesting - it took me a minute to get how this was working because the `toMatchSnapshot()` seemed a little unclear at first, but I see now that the snapshot holds a bunch of states keyed by the `it` statement. Nice.

Reply:
Exactly, it's very helpful. We can catch errors easily with this.